app/controllers/taxonomy_controller.rb

```diff
@@ -0,0 +1,44 @@
+class TaxonomyController < ApplicationController
+
+  layout :determine_layout
+
+  def index
+    initialize_taxonomy
+  end
+
+  def categories
```

Comment:
use just the index action, but add a parameter like `section=categories` or `section=groups`